app/helpers/mentions_helper.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module MentionsHelper
+  def mention_matcher
+    @mention_matcher ||= begin
+      users = current_project.authors.all.select(:email, :name)
```

Comment:
Do we need this extra call?

`current_project.authors` is already cached as `@mentionable_users` by `app/controllers//concerns/mentioned.rb` and also included as a `<meta>` tag in the layout's header via `app/views//comments/_mentionable_users.html.erb`.

Reply:
I had higher hopes for the `meta` tag. But because we're doing the parsing server side it doesn't help unless we opt to move the mention rendering to the client...

Having the `@mentionable_users` already cached could be utilized, but we would need to expand it to be included in `app/controllers/comments_controller.rb` and in the create, update actions, not just show.

Because the comments get rendered 1 by 1 when being created or edited we need the list accessible at that point.

It makes it explicit when and where mentions can be rendered. Where right now any call to the mentions formatter will prep and cache for the request. I'll expand Mentioned to support other actions (when requested) add a commit and see what you think.

Reply:
I knew there was an outlier. `activites_controller` also renders comments, which means we would need it there too which would cache the query on every poll interval, even if it doesn't have comments, or need mentions.

Maybe there's an inbetween- although it can be dangerous for performance if we miss it later.

`users = @mentionable_users ? @mentionable_users : current_project.authors.all.select(:email, :name)`

This will use the cached collection if available, but also build a new one only if needed.